### PR TITLE
329 email final edit and comparison

### LINKED ISSUE #329: Email final edit and comparison
The current state of the emails is that they are normalized but the information needs final review.

* Names have not yet been implemented in the emails
* Document type had odd formatting (i.e. doctoral_dissertation)
* Did not get the "you started the submission email"

See #276

### PR DESCRIPTION
**Pull request by jeremyf**
## Claiming issue 329

@a3e3e657b76f6b3b005c5398d8b7393fee23cde0

relates to #329

$ ./script/start-issue 329

[skip ci]

## Renaming email notification

@37fd2a03cd4c638388b43d62efaca7659c696e60

From confirmation_of_entity_created to confirmation_of_work_created.

The email template is very ETD specific, but that will have to do for
now.

## Inching towards completed grad school signoff

@beb9fca030788cbd32bfec4047886ffbb4e74c86

It appears as though we will need to be capturing additional
information. However, instead of holding this issue hostage as that
feature is worked on, I want to move forward with a round of
revisions.

Related to #421

## Refining email templates for clarity

@97267608bed0794b05b6e6b541d096cfd99dcaf3

## Fixing date rendering to not be UTC

@76bdd19b2c7f2aa4166ac4aa5160f729268cf72c

Seriously, no body wants to see the UTC code. If we have a date it
will render as: `November 10th, 2007`. If we have a time it will
render as: `December 4th, 2007 12:31`. I assume it will render hours
as "13:45" instead of "1:45pm".

## Tidying up advisor sign off is complete

@f3e1173f060f49745fa8eb2cb5a0903ebcd56cae

Instead of an ugly rendered definition list, I'm opting for a more
compact message.

## Adding Collaborator#to_s

@eba8b594e51e57ace5679e0d45b0b0b607322dc2

Because this is a friendly thing to do.

## Passing a comment to the notifier

@c6a0f6a13afba608be98a28dd104093ffb69bbeb

In the live demonstration, an exception was raised because the emailer
was assuming a processing comment when it in fact received a work.

## Ensuring that work type for emails is titleized

@53fb68eccd4496a056b448581b4b9c7c41d99520

This knowledge is repeated through-out the application. Look at a
means of normalizing the work_type method to be titleized.

## Switching p-tag for comment to blockquote

@ce0f0157007b810a8d7b2d9019d99e9bc0c598e0

Its more semantic, and helps tidy up the email.

## Extracting common comments behavior

@a534bd0858ae46c062e8336e6281b8065ea8342b

I don't know if this is over normalization, but there was a
duplication of semantics and meaning.

## Extracting a partial for accessible objects

@27cb23a79734fa18dcb53b72d9a812caaae93a75

The logic is analogous to what is occurring in the controller view; So
I'm attempting to normalize this.

## Adding url_host and specs for Dragonfly

@0d5c1413fc5a2884ca89252078e00b3e76375a23

Prior to adding a url_host to Dragonfly, the Attachment#file_url was
returning a path. This resolves that.

I'm not certain, as we proceed, if this will ease or hinder other
developments.

## Preferring Figaro over application secrets

@7788d954e7d28480b986b1ac37c1ad9a92aef53d

## Specifying a more meaningful local host

@95bf0ceebd244a10cd4506b48dba5c97cbf936ff

Instead of the dubious example.com, keep it local.

## Addressing SimpleForm deprecation warning

@02c7518d0425de82f9d7e547030d2f0c6d0ea9ff

```console
DEPRECATION WARNING: [SIMPLE_FORM] SimpleForm.form_class= is
deprecated and will be removed in 4.x. Use
SimpleForm.default_form_class= instead. (called from block in
<top (required)> at RAILS_ROOT/config/initializers/simple_form.rb:101)
```

## Adding explicit URL host declaration

@5fa8c18ff9cb8a1294cb4356e49a46e60f4f747c

Instead of the obtuse scheme + domain_name

## Adding a bit more configurability to access_url

@10cac972b95756ece6f385859f1fa659352d720a

By passing the template, I can leverage the email template engine vs
the view template engine. I noticed a difference when I failed to
configure the email's base url for notifications that were sent.

## Verifying that Attachment#file_url is a url

@6ae8db9ff7e8149d2c6064092a18168541eb2104

Because later on I'm assuming that #file_url is a URL instead of a
path, I want to make sure that the configuration is correct.

This is perhaps an impossibility.

## Normalizing accessible objects rendering

@ce175a7eb8800fcccf3b4142055b059c39537cd1

Instead using one that was custom for the mailer, I'm opting to use
the generalized html for rendering the accessible objects.

Note: I need to use the #to_s method because action mailer has a
different rendering template.

## Reverting from permanent URI to the action URL

@efbfeefaf08a4817658b45117c0056853f4856b2

This is a stop gap as I would prefer to use the permanent URI. They
are the same things but it is not obvious that they are the same
thing.

## Collapsing multi-lined method call

@8d7389db3b5008943d8e4081dcfdf5f6a7104db5

This was making rubocop angry; And was unnecessary.

## Closing issue 329

@bf98c07efee27d31132a3e77b0e6998ad209426d

Closes #329

$ ./script/close-issue 329

### CODE REVIEW

File: app/views/sipity/mailers/email_notifier/confirmation_of_work_created.html.erb [hunk 1]
```diff
@@ -1,13 +1,13 @@
 <p>Welcome to the Electronic Thesis and Dissertation (ETD) submission site. You have begun a formal submission for the following document:</p>
 
-<dl>
-  <dd>Title: </dd> <dt><%= @entity.title %></dt>
-  <dd>Submission Type: </dd> <dt><%= @entity.work_type %></dt>
-</dl>
+<ul>
+  <li><strong>Title:</strong><%= @entity.title %></li>
+  <li><strong>Submission Type:</strong><%= @entity.work_type %></li>
```
Comment (danhorst) [inline app/views/sipity/mailers/email_notifier/confirmation_of_work_created.html.erb]: There should be a space after the closing strong tag.

File: app/views/sipity/mailers/email_notifier/confirmation_of_advisor_signoff.html.erb [hunk 2]
```diff
@@ -1,4 +1,4 @@
-<p>On <%= @entity.action_taken_at %>, <%= @entity.on_behalf_of %> approved <%= link_to "your #{@entity.work_type}", @entity.email_message_action_url %> for submission to the Graduate School.</p>
+<p>On <%= @entity.action_taken_at.to_formatted_s(:long_ordinal) %>, <%= @entity.on_behalf_of %> approved <%= link_to "your #{@entity.work_type}", @entity.email_message_action_url %> for submission to the Graduate School.</p>
```
Comment (danhorst) [inline app/views/sipity/mailers/email_notifier/confirmation_of_advisor_signoff.html.erb]: I'm getting a blank response for `@entity.on_behalf_of`
screen shot 2015-04-02 at 9 53 38 am

File: app/views/sipity/mailers/email_notifier/advisor_signoff_is_complete.html.erb [hunk 3]
```diff
@@ -1,19 +1,11 @@
...
-          <dd><%= advisor.name %></dd>
-          <dt>NetID:</dt>
-          <dd><%= advisor.netid %></dd>
-          <dt>Email:</dt>
-          <dd><%= advisor.email %></dd>
-          <dt>Role:</dt>
-          <dd><%= advisor.role %></dd>
-        </dl>
+        <strong><%= advisor.role %>:</strong>
+        <%= advisor.name %> <%= "(NetID: #{mail_to(advisor.netid + '@nd.edu', advisor.netid)})" if advisor.netid.present? %> <%= "(Email: #{mail_to(advisor.email)})" if advisor.email.present? %>
```
Comment (danhorst) [inline app/views/sipity/mailers/email_notifier/advisor_signoff_is_complete.html.erb]: The mailto link has not been marked as `html_safe`
screen shot 2015-04-02 at 9 58 55 am

File: app/views/sipity/mailers/email_notifier/confirmation_of_advisor_signoff.html.erb (hunk 2 above)
Comment (jeremyf) [inline app/views/sipity/mailers/email_notifier/confirmation_of_advisor_signoff.html.erb]: The `<%= @entity.on_behalf_of %>` has been addressed by "Adding a fallback hack for User#to_s to username" (commit @8b9e343)

**Pull request merged**